lib/connectors/SentinelConnector/index.ts

```diff
@@ -269,19 +294,108 @@ export default class SentinelConnector extends AbstractConnector {
       commandTimeout: this.options.sentinelCommandTimeout,
       dropBufferSupport: true,
     });
+  }
+
+  private async resolve(
+    endpoint: Partial<ISentinelAddress>
+  ): Promise<ITcpConnectionOptions | null> {
+    const client = this.connectToSentinel(endpoint);
 
     // ignore the errors since resolve* methods will handle them
     client.on("error", noop);
 
+    let result: ITcpConnectionOptions | null = null;
+
     try {
       if (this.options.role === "slave") {
-        return await this.resolveSlave(client);
+        result = await this.resolveSlave(client);
       } else {
-        return await this.resolveMaster(client);
+        result = await this.resolveMaster(client);
       }
+
+      if (result) {
+        this.initFailoverDetector({
+          address: endpoint,
+          isConnected: true,
```

Comment:
As I mentioned in the last comment, the connection should be pretty lightweight (especially given this is a one-timething) so we can just disconnect here and create a new one in `initFailoverDetector` for better code readability. No strong opinion though 😄